modelon/impact/client/entities/custom_function.py

```diff
@@ -116,16 +111,30 @@ def parameter_values(self) -> Dict[str, Any]:
         """Custom_function parameters and value as a dictionary"""
         return {p.name: p.value for p in self._param_by_name.values()}
 
-    def _get_options(self, use_defaults):
+    def get_options(self, use_defaults: Optional[bool] = False):
+        """Get project execution option.
+
+        Parameters:
+
+            use_defaults --
+                If true, default compiler options are used.
+
+        Example::
+            opts = custom_function.get_compiler_options()
+            opts_2 = opts.compiler_options.with_values(c_compiler='gcc')
+        """
         if use_defaults:
-            return self._custom_func_sal.custom_function_default_options_get(
+            options = self._sal.custom_function.custom_function_default_options_get(
                 self._workspace_id, self._name
             )
-        return self._custom_func_sal.custom_function_options_get(
-            self._workspace_id, self._name
-        )
+        else:
+            options = self._sal.custom_function.custom_function_options_get(
+                self._workspace_id, self._name
+            )
+        options['customFunction'] = self.name
+        return ProjectExecutionOptions(options)
```

Comment:
It would make more sense to have this as:
`return ProjectExecutionOptions(options, self.name)`

and from the place where this logic originates:
`ProjectExecutionOptions(execution_option, execution_option['customFunction'])`